### ISSUE TITLE
SC2242: inflated severity for passing negative numbers to exit in bash

### ISSUE THREAD

**Issue by pabs3** [koalaman/shellcheck#1952]
#### For bugs

- Rule Id: SC2242
- My shellcheck version: 0.7.1
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)

I could not try on shellcheck.net due to the JavaScript requirement

#### Here's a snippet or screenshot that shows the problem:

```sh
exit -1
```

#### Here's what shellcheck currently says:

The severity of passing negative numbers to exit is error for both bash and POSIX shell:

```
$ shellcheck --severity error -s sh foo.sh 

In foo.sh line 1:
exit -1
     ^-- SC2242: Can only exit with status 0-255. Other data should be written to stdout/stderr.

For more information:
  https://www.shellcheck.net/wiki/SC2242 -- Can only exit with status 0-255. ...
```

```
$ shellcheck --severity error -s bash foo.sh 

In foo.sh line 1:
exit -1
     ^-- SC2242: Can only exit with status 0-255. Other data should be written to stdout/stderr.

For more information:
  https://www.shellcheck.net/wiki/SC2242 -- Can only exit with status 0-255. ...
```

#### Here's what I wanted or expected to see:

I expected that the severity should depend on the shell in use and the parameter passed to exit. bash translates negative numbers into positive numbers 0-255 but other shells do not. So SC2242 should only be an error for dash and POSIX shell (and possibly ksh, untested).

```
$ shellcheck --severity error -s sh foo.sh 

In foo.sh line 1:
exit -1
     ^-- SC2242: Can only exit with status 0-255. Other data should be written to stdout/stderr.

For more information:
  https://www.shellcheck.net/wiki/SC2242 -- Can only exit with status 0-255. ...
```

```
$ shellcheck --severity error -s bash foo.sh 
```

#### Here is what various shells do:

```
$ busybox sh foo.sh ; echo $?
foo.sh: exit: line 1: Illegal number: -1
2
```

```
$ dash foo.sh ; echo $?
foo.sh: 1: exit: Illegal number: -1
2
```

```
$ bash foo.sh ; echo $?
255
```

The version details:

```
$ busybox | head -1
BusyBox v1.30.1 (Debian 1:1.30.1-4) multi-call binary.
$ dpkg -l dash | tail -n1
ii  dash           0.5.10.2-7   amd64        POSIX-compliant shell
$ bash --version | head -1
GNU bash, version 5.0.16(1)-release (x86_64-pc-linux-gnu)
```
